src/Microsoft.AspNetCore.Sockets/HttpConnectionDispatcher.cs

```diff
@@ -378,6 +359,28 @@ private ConnectionState CreateConnection(HttpContext context)
             }
         }
 
+        private async Task<IList<Message>> ExtractMessages(HttpContext context, byte[] buffer)
```

Comment:
This way the code is magic because I would report a detailed error in a different place that I detected it. It may work initially but as code evolves it will be hard to keep the two in sync. Also, we already have methods which return `Task<T>` for exactly the same reason I did - writing to the response in case of error:  https://github.com/aspnet/SignalR/blob/dev/src/Microsoft.AspNetCore.Sockets/HttpConnectionDispatcher.cs#L407